apiserver/facades/agent/upgrader/register.go

```diff
@@ -41,55 +49,55 @@ func newUpgraderFacade(ctx facade.ModelContext) (Upgrader, error) {
 	if err != nil {
 		return nil, apiservererrors.ErrPerm
 	}
+
+	st := ctx.State()
 	model, err := st.Model()
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 
+	domainServices := ctx.DomainServices()
+
+	if tag.Kind() == names.UnitTagKind && model.Type() != state.ModelTypeCAAS {
+		return NewUnitUpgraderAPI(
+			st,
+			auth,
+			domainServices.Agent(),
+			ctx.WatcherRegistry(),
+			domainServices.Application(),
```

Comment:
My OCD wants all the domain services grouped together, YMMV.

Reply:
fair